js/radio/listen.directive.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+(function(){
+  angular
+  .module("songs")
+  .directive("songListen", [
+    "$state",
+    SongListenDirectiveFunction
+  ])
+  function SongListenDirectiveFunction($state){
+    return{
+      templateUrl: "js/radio/listen.html",
+      scope: {
+        song: "="
+      },
+      link: function(scope){
+        scope.song.url = "{{SongShowViewModel.song.audio_url}}";
```

Comment:
do you get any errors with this line? also can probably remove the quotes and curlies, and pass the current song in as an attribute